crates/fluvio-hub-util/src/fvm/api.rs

```diff
@@ -0,0 +1,116 @@
+//! Hub FVM API Client
+
+use anyhow::{Error, Result};
+use surf::get;
+use url::Url;
+
+use super::{Channel, PackageSet, RustTarget};
+
+/// Base URL for `fmv` related API requests
+pub const API_BASE_URL: &str = "https://hub.infinyon.cloud/hub/v1/fvm";
+
+/// HTTP Client for interacting with the Hub FVM API
+pub struct Client {
```

Comment:
can we reuse same client instead of duplicating?

Reply:
I did it like this to be flexible for development as we work on Hub's dev instance as well.

By merging FVM API Client with `HubAccess` client we would be inheriting authentication protocols from the Hub Access which are not relevant for FVM given that FVM is a public API and Hub Access requires authentication and token signing.

```
    pub fn new() -> Self {
        HubAccess {
            remote: HUB_REMOTE.to_string(),
            hubid: String::new(),
            pkgkey: String::new(),
            pubkey: String::new(),
        }
    }
```

I could move the URI to the `make_url...` methods as well and reuse `HUB_REMOTE` URL as the de-facto base URL.